clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventHandler.java

```diff
@@ -42,26 +44,32 @@
 public class ApplicationEventHandler implements Closeable {
 
     private final Logger log;
+    private final Time time;
     private final BlockingQueue<ApplicationEvent> applicationEventQueue;
```

Comment:
I wonder if we could actually make a class for the application event queue, rather than using `BlockingQueue` directly. In this way, we could do things like incorporate the queue size metric into the queue, rather than having update the metrics in the caller.

Reply:
I think `ApplicationEventHandler` is a class to encapsulate application event queue. However, it couples with `ConsumerNetworkThread`, so we can't just add the draining event function to `ApplicationEventHandler`. How about we solve it in followup Jira https://issues.apache.org/jira/browse/KAFKA-18048? We had related discussion in previous comment https://github.com/apache/kafka/pull/17199#discussion_r1849058419.

Reply:
I agree that this is an area that could use some refactoring. Thanks for filing KAFKA-18048, @FrankYang0529.